packages/firebase_auth/firebase_auth/example/test_driver/firebase_auth_e2e.dart

```diff
@@ -4,29 +4,47 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-// @dart=2.9
-
 import 'package:drive/drive.dart' as drive;
-//import 'package:firebase_auth/firebase_auth.dart'; // only needed if you use the Auth Emulator
+import 'package:firebase_auth/firebase_auth.dart';
 import 'package:firebase_core/firebase_core.dart';
 import 'package:flutter_test/flutter_test.dart';
 
 import 'instance_e2e.dart';
+import 'test_utils.dart';
 import 'user_e2e.dart';
 
 // Requires that an emulator is running locally
-bool USE_EMULATOR = false;
+// `melos run firebase:emulator`
+bool useEmulator = true;
 
 void testsMain() {
   setUpAll(() async {
     await Firebase.initializeApp();
+    if (useEmulator) {
+      await FirebaseAuth.instance
+          .useEmulator('http://$testEmulatorHost:$testEmulatorPort');
+    }
+  });
 
-    // Configure the Auth test suite to use the Auth Emulator
-    // This may not be enabled until the test suite is ported to:
-    //  - have ability to create disabled users
-    //  - have ability to fetch OOB and SMS verification codes
-    // JS implementation to port to dart here: https://github.com/invertase/react-native-firebase/pull/4552/commits/4c688413cb6516ecfdbd4ea325103d0d8d8d84a8#diff-44ccd5fb03b0d9e447820032866f2494c5a400a52873f0f65518d06aedafe302
-    // await FirebaseAuth.instance.useEmulator('http://localhost:9099');
+  setUp(() async {
```

Comment:
Just verifying this is run before each individual test? I found a few cross-test dependencies when I moved to per-test-reset-recreate so it was worthwhile I think. I believe that's what is happening here and if so +1

Reply:
Aye this is before each test, [`setUp`](https://api.flutter.dev/flutter/flutter_test/setUp.html) vs [`setUpAll`](https://api.flutter.dev/flutter/flutter_test/setUpAll.html)